Gems/GradientSignal/Code/Include/GradientSignal/Ebuses/GradientRequestBus.h

```diff
@@ -49,6 +51,29 @@ namespace GradientSignal
         */
         virtual float GetValue(const GradientSampleParams& sampleParams) const = 0;
 
+        /**
+         * Given a list of positions, generate values. Implementations of this need to be thread-safe without using locks,
+         * as it can get called from multiple threads simultaneously and has the potential to cause lock inversion deadlocks.
+         * \param positions The input list of positions to query.
+         * \param outValues The output list of values. This list is expected to be the same size as the positions list.
+         */
+        virtual void GetValues(AZStd::array_view<AZ::Vector3> positions, AZStd::array_view<float> outValues) const
+        {
+            // Reference implementation of GetValues for any gradients that don't have their own optimized implementations.
+            // This is 10%-60% faster than calling GetValue via EBus many times due to the per-call EBus overhead.
+            AZ_Assert(positions.size() == outValues.size(), "input and output lists are different sizes.");
+            GradientSampleParams sampleParams;
+            for (size_t index = 0; index < positions.size(); index++)
+            {
+                sampleParams.m_position = positions[index];
```

Comment:
Fair point, I've added a safety check.  I'm not adding a return code for failures though or extra code to make it graceful, since it's a programming error, and we'd either need that extra code on every override of GetValues too or else additional complications to the API (GetValuesInternal) just for common handling of the error condition.